test/trust-proxy.test.js

```diff
@@ -169,3 +169,22 @@ test('trust proxy IP addresses', (t) => {
     sgetForwardedRequest(app, '3.3.3.3, 2.2.2.2, 1.1.1.1', '/trustproxyipaddrs')
   })
 })
+
+test('validation trust proxy chain', (t) => {
```

Comment:
Could you explain to me this test?
I don't understand why in this case we don't want to read the `X-Forwarded-Host`
The `1.1.1.1` forward is trusted, but don't we trust the `127.0.0.1` client?

Reply:
We need ensures that only the IPs on the chain will be trusted

The request socket connection is 127.0.0.1 that IP, not in the chain, in that case, 
We want to ignore from headers:
X-Forwarded-Host, X-Forwarded-For, and  X-Forwarded-Proto

So that test checks if the IP did not change to 1.1.1.1 (X-Forwarded-For) 
And the hostname did not change to 'example.com' (X-Forwarded-Host)